src/qibo/backends/npmatrices.py

```diff
@@ -17,51 +17,53 @@ def _cast(self, x, dtype):
 
     @cached_property
     def H(self):
-        return self.np.array([[1, 1], [1, -1]], dtype=self.dtype) / self.np.sqrt(2)
+        return self._cast([[1, 1], [1, -1]], dtype=self.dtype) / self.np.sqrt(2)
 
     @cached_property
     def X(self):
-        return self.np.array([[0, 1], [1, 0]], dtype=self.dtype)
+        return self._cast([[0, 1], [1, 0]], dtype=self.dtype)
 
     @cached_property
     def Y(self):
-        return self.np.array([[0, -1j], [1j, 0]], dtype=self.dtype)
+        return self._cast([[0j, -1j], [1j, 0j]], dtype=self.dtype)
 
     @cached_property
     def Z(self):
-        return self.np.array([[1, 0], [0, -1]], dtype=self.dtype)
+        return self._cast([[1, 0], [0, -1]], dtype=self.dtype)
 
     @cached_property
     def SX(self):
-        return self.np.array([[1 + 1j, 1 - 1j], [1 - 1j, 1 + 1j]], dtype=self.dtype) / 2
+        return self._cast([[1 + 1j, 1 - 1j], [1 - 1j, 1 + 1j]], dtype=self.dtype) / 2
 
     @cached_property
     def SXDG(self):
-        return self.np.transpose(self.np.conj(self.SX))
+        return self._cast([[1 - 1j, 1 + 1j], [1 + 1j, 1 - 1j]], dtype=self.dtype) / 2
 
     @cached_property
     def S(self):
-        return self.np.array([[1, 0], [0, 1j]], dtype=self.dtype)
+        return self._cast([[1 + 0j, 0j], [0j, 1j]], dtype=self.dtype)
 
     @cached_property
     def SDG(self):
-        return self.np.conj(self.S)
```

Comment:
Same about the `conj`. Here I am not sure if there is really an issue because `np.conj` is being used in some other matrices (ie in `CSXDG`), maybe you need to modify those as well?

Reply:
Yes, all matrices would have to be hard-coded instead of relying in any `numpy` function